packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1677,12 +1737,23 @@ export class Cluster extends ClusterBase {
       new CfnOutput(this, 'ClusterName', { value: this.clusterName });
     }
 
+    const supportAuthenticationApi = (this.authenticationMode === AuthenticationMode.API ||
+      this.authenticationMode === AuthenticationMode.API_AND_CONFIG_MAP) ? true : false;
+
     // do not create a masters role if one is not provided. Trusting the accountRootPrincipal() is too permissive.
     if (props.mastersRole) {
       const mastersRole = props.mastersRole;
 
-      // map the IAM role to the `system:masters` group.
-      this.awsAuth.addMastersRole(mastersRole);
+      // if we support authentication API we create access entry
+      if (supportAuthenticationApi) {
+        this.grantAccess('mastersRoleAccess', props.mastersRole.roleArn, [
+          AccessPolicy.fromAccessPolicyName('AmazonEKSClusterAdminPolicy'),
```

Comment:
Is it a good idea to add Admin Policy by default?

Reply:
In current aws-eks implementation, if `props.mastersRole` is defined, we add that IAM principal to AwsAuth as admin role using [addMastesrRole method](https://github.com/aws/aws-cdk/blob/205163fc0d2cac84d3d746a98c393e137f0e2388/packages/aws-cdk-lib/aws-eks/lib/aws-auth.ts#L64), which essentially create the classic ConfigMap with that `props.mastersRole` as `system:masters`, and that is essentially what we'd like it to happen using the new AccessEntry approach - to grant that principal the admin policy

https://github.com/aws/aws-cdk/blob/205163fc0d2cac84d3d746a98c393e137f0e2388/packages/aws-cdk-lib/aws-eks/lib/cluster.ts#L1680-L1685

But you are right, it's not clear if we should grant it `AmazonEKSAdminPolicy` or `AmazonEKSClusterAdminPolicy` and I can't find it in our doc.

This is what Amazon Q told me:

<img width="457" alt="image" src="https://github.com/aws/aws-cdk/assets/278432/0f17329a-22e3-4c55-b33d-d7172e85c199">

I'll cut an internal ticket to the team to get this clarified.

Reply:
I guess masters role on the K8s cluster is meant for granting the cluster full access, but i'm not confident about the difference between `AmazonEKSClusterAdminPolicy` and `AmazonEKSAdminPolicy`, and which one to use here.

Reply:
`AmazonEKSClusterAdminPolicy` should have higher privileges than `AmazonEKSAdminPolicy` as it's cluster-wise while `AmazonEKSAdminPolicy` is per namespace(s). I'll cut a ticket to get it sorted.

Reply:
Internal Tracking: V1385204928

Reply:
internal tracking: V1385204928